### ISSUE TITLE
Unclear what it means for a room to be "public"

### ISSUE THREAD

**Issue by DMRobertson** [matrix-org/matrix-spec#1144]
**Link to problem area**: https://spec.matrix.org/v1.3/client-server-api/#listing-rooms

**Issue**

My current understanding:
1. the room directory shows a list of rooms deemed to be public
2. rooms can be included or excluded from this list via `PUT /_matrix/client/v3/directory/list/room/{roomId}`
3. I think(?) there is some relationship between join rules and this list. But the details aren't specified?
  - Are rooms with join_rules `public` included in this list, unless they've been opted out?
  - What about `knock` and `knock_restricted`?
  - Can you try to mark an `invite-only` room as public?
  - Does the homeserver have some discretion here?

In a nutshell, I'd like to see a short bit of prose which explains that the join rules and visibility are separate (but not orthogonal?) concepts.

**Comment by MTRNord** [comment]
In addition to that: does a room need to allow guest access to be "public"?

**Comment by babolivier** [comment]
For clarity, the current implementation in Synapse is to only change the status of a room in the directory based on requests to `PUT /_matrix/client/v3/directory/list/room/{roomId}` and the `visibility` attribute at room creation. So no relationship with join rules, guest access etc, but I agree the spec is a bit confusing and unclear on that point.

**Comment by richvdh** [comment]
https://github.com/matrix-org/matrix-spec/issues/633

**Comment by anoadragon453** [comment]
And https://github.com/matrix-org/matrix-spec/issues/83.

**Comment by anoadragon453** [comment]
I'm going to consider this a duplicate of the above two issues.
